sos/plugins/openvswitch.py

```diff
@@ -95,7 +107,10 @@ def setup(self):
             # Capture DPDK pmd stats
             "ovs-appctl dpif-netdev/pmd-stats-show",
             # Capture DPDK pmd performance counters
-            "ovs-appctl dpif-netdev/pmd-perf-show"
+            "ovs-appctl dpif-netdev/pmd-perf-show",
+            # Collect ovs-vswitchd thread details in ps command.
+            "ps -p %s -Lo pid,tid,user,pcpu,psr,trs,rss,size,sz,drs,vsz,\
+maj_flt,min_flt,class,rtprio,ni,pri,s,stat,wchan,flags,comm" % pid
```

Comment:
Can this be indented to stay with sos-style?

Reply:
Adding space between ps options will break the ps command. 
Also the ps options are so long that having them in one line will conflict pep8. Would that be okay? Eg:

            "ps -p %s -Lo \
            pid,tid,user,pcpu,psr,trs,rss,size,sz,drs,vsz,maj_flt,min_flt,class,rtprio,ni,pri,s,stat,wchan,flags,comm" \
            % pid

Reply:
You just need to build the final argument string piece-wise: there are various examples of this in sos already, e.g. from the `process` plugin:

```python
        ps_axo = "ps axo"
        # process group and thread options
        ps_group_opts = "pid,ppid,user,group,lwp,nlwp,start_time,comm,cgroup"
        ps_sched_opts = "flags,state,uid,pid,ppid,pgid,sid,cls,pri,addr,sz,"
        ps_sched_opts += "wchan:20,lstart,tty,time,cmd"
[...]
        self.add_cmd_output([
            "ps auxwwwm",
            "ps alxwww",
            "ps -elfL",
            "%s %s" % (ps_axo, ps_group_opts),
            "%s %s" % (ps_axo, ps_sched_opts)
        ])
```

That said, I'm not a huge fan of spreading more `ps` collection around different plugins if it can be avoided - there is a great deal over overlap here between existing data that's collected.

Reply:
I dont mind adding these options to process plugin's ps_sched_opts, if you find that should be alright.